core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -613,6 +649,17 @@ public void onActivityStarted(@NonNull Activity activity) {
                 if (getOptStatus()) {
                     return;
                 }
+                // Session Tracking
+                // Automatic tracking session started
+                if (config.isTrackAutoSession()) {
+                    if (!applicationInstalledOrUpdated) {
+                        if (RudderClient.userSession != null) {
```

Comment:
Can we start the session as well in the application opened itself, instead of doing it in both Application Installed and Application Updated ?

Reply:
We are doing it in `checkSessionTimeoutDuration`
If we create a session only in `Application Opened `event, `Application Installed` & `Application Updated` will miss out on the session.

Reply:
okay got that, then in that case can we do it in the SDK initialization itself, before checking for Application Installed and Application Updated ?

Reply:
Addressed them